src/main/java/mat/server/fhirvalidation/FhirValidationReportServlet.java

```diff
@@ -14,50 +16,74 @@
 
     @Override
-    protected void doGet(HttpServletRequest req, HttpServletResponse response) throws IOException {
-        String sessionId = req.getSession().getId();
-        String id = req.getParameter(ID_PARAM);
-        boolean converted = isConverted(req);
-        logger.info("Generating validation report for measure id: " + id + " converted: " + converted);
-        boolean showStackTrace = isShowStackTrace(req);
-        String validationReport = "";
+    protected void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+
         try {
-            VsacTicketInformation vsacTicketInformation = getVascApiService().getTicketGrantingTicket(sessionId);
-            if (vsacTicketInformation == null) {
-                throw new MatException("Cannot get a granting ticket");
-            }
-            validationReport = getValidationReportService().getFhirConversionReportForMeasure(id, vsacTicketInformation.getTicket(), converted);
+            var sessionId = request.getSession().getId();
+            var ticket = getVsacTicket(sessionId);
+
+            var converted = isConverted(request);
+            var id = request.getParameter(ID_PARAM);
+
+            var validationReportService = getFhirValidationReportService(request);
+            var validationReport = validationReportService.generateReport(id, ticket, converted);
+
+            response.setHeader(CONTENT_TYPE, MediaType.TEXT_HTML_VALUE);
+            response.getOutputStream().write(validationReport.getBytes());
         } catch (Exception e) {
-            logger.error("Exception occurred while generation FHIR conversion report:", e);
-            validationReport = "An error occurred while validating the FHIR conversion. Please try again later. " +
-                    "If this continues please contact the mat help desk.";
-            if (showStackTrace) {
-                StringBuilder debugInfo = new StringBuilder(validationReport);
-                debugInfo
-                        .append(" Error ")
-                        .append(System.lineSeparator())
-                        .append(e.getMessage())
-                        .append(ExceptionUtils.getStackTrace(e));
-                validationReport = debugInfo.toString();
-            }
+            var errorResponse = toErrorResponse(request, e);
+            response.setHeader(CONTENT_TYPE, MediaType.TEXT_HTML_VALUE);
             response.setStatus(HttpServletResponse.SC_INTERNAL_SERVER_ERROR);
+            response.getOutputStream().write(errorResponse.getBytes());
+        }
+
+
+    }
+
+    /**
+     * Given sessionId, delegates to VsacApiService to fetch the VSAC ticket
+     * @param sessionId
+     * @return vsac Ticket
+     * @throws MatException
+     */
+    private String getVsacTicket(String sessionId) throws MatException {
+        var vsacTicketInformation = getVascApiService().getTicketGrantingTicket(sessionId);
```

Comment:
To Validate/convert Standalone Cql library, we don't need Vsac token. Can you Confirm with Michael ?

Reply:
Right now we don't, but in stories yet to come we will need it again. Please leave it in there.

Reply:
Gotcha